pkg/operator/sync.go

```diff
@@ -119,30 +119,33 @@ func isCloudConfigRequired(infra *configv1.Infrastructure) bool {
 
 // Sync cloud config on supported platform from cloud.conf available in openshift-config-managed/kube-cloud-config ConfigMap.
 func (optr *Operator) syncCloudConfig(spec *mcfgv1.ControllerConfigSpec, infra *configv1.Infrastructure) error {
-	if _, err := optr.clusterCmLister.ConfigMaps("openshift-config-managed").Get("kube-cloud-config"); err != nil {
+	cm, err := optr.clusterCmLister.ConfigMaps("openshift-config-managed").Get("kube-cloud-config")
+	if err != nil {
 		if apierrors.IsNotFound(err) {
 			if isCloudConfigRequired(infra) {
-				// Return error only if cloud config is required, otherwise prceeds further.
+				// Return error only if cloud config is required, otherwise proceeds further.
 				return fmt.Errorf("%s/%s configmap is required on platform %s but not found: %v",
 					"openshift-config-managed", "kube-cloud-config", infra.Status.PlatformStatus.Type, err)
 			}
-		} else {
-			return err
+			return nil
 		}
-
-	} else {
-		// Read cloud.conf from openshift-config-managed/kube-cloud-config ConfigMap.
-		cc, err := optr.getCloudConfigFromConfigMap("openshift-config-managed", "kube-cloud-config", "cloud.conf")
-		if err != nil {
-			return err
+		return err
+	}
+	// Read cloud.conf from openshift-config-managed/kube-cloud-config ConfigMap.
+	cc, err := getCloudConfigFromConfigMap(cm, "cloud.conf")
+	if err != nil {
+		if isCloudConfigRequired(infra) {
+			// Return error only if cloud config is required, otherwise proceeds further.
+			return fmt.Errorf("%s/%s configmap must have the %s key on platform %s but not found",
+				"openshift-config-managed", "kube-cloud-config", "cloud.conf", infra.Status.PlatformStatus.Type)
 		}
-
+	} else {
 		spec.CloudProviderConfig = cc
+	}
 
-		caCert, err := optr.getCAsFromConfigMap("openshift-config-managed", "kube-cloud-config", "ca-bundle.pem")
-		if err == nil {
-			spec.CloudProviderCAData = caCert
-		}
+	caCert, err := getCAsFromConfigMap(cm, "ca-bundle.pem")
```

Comment:
One question, do we know that non-aws platform will make use of kube-cloud-config cm to just keep `ca-bundle.pem` data?

Reply:
I do not quite understand your question.

Currently, AWS and OpenStack are the only two platforms that make use of the `ca-bundle.pem`.

Reply:
Ah ok, that's what I wanted to know that whether multiple platforms are making using of ca-bundle.pem key without having cloud.conf key in kube-cloud-config cm.